stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplReactionsTest.kt

```diff
@@ -131,12 +144,77 @@ internal class ChannelControllerImplReactionsTest {
             result.latestReactions.contains(deletedReaction) `should be equal to` false
         }
 
+    @Test
+    fun `when deleting a reaction while offline, status must be right and reaction inserted`() =
+        runBlockingTest {
+            val sut = Fixture(testCoroutines.scope, currentUser)
+                .givenMockedRepositories()
+                .givenMessageWithReactions(myReactions)
+                .get()
+
+            whenever(sut.domainImpl.isOnline()) doReturn false
+
+            val result = sut.deleteReaction(mockReaction)
+
+            inOrder(mockReaction) {
+                verify(mockReaction).syncStatus = SyncStatus.IN_PROGRESS
+                verify(mockReaction).syncStatus = SyncStatus.SYNC_NEEDED
+            }
+
+            result.isSuccess `should be equal to` true
+
+            verify(sut.domainImpl.repos).insertReaction(mockReaction)
+        }
+
+    @Test
+    fun `when delete reaction, the sync status is right and reaction is inserted`() =
+        runBlockingTest {
+            val sut = Fixture(testCoroutines.scope, currentUser)
+                .givenMockedRepositories()
+                .givenMessageWithReactions(myReactions)
+                .get()
+
+            whenever(sut.domainImpl.isOnline()) doReturn true
+            whenever(sut.domainImpl.runAndRetry<Message>(any())) doAnswer { Result(ChatError()) }
+
+            val result = sut.deleteReaction(mockReaction)
+
+            inOrder(mockReaction) {
+                verify(mockReaction).syncStatus = SyncStatus.IN_PROGRESS
+                verify(mockReaction).syncStatus = SyncStatus.SYNC_NEEDED
```

Comment:
I think it would be nice to add an additional test for `SyncStatus.FAILED_PERMANENTLY` case